src/idl_gen_js.cpp

```diff
@@ -110,16 +144,23 @@ class JsGenerator : public BaseGenerator {
   std::string &exports = *exports_ptr;
   for (auto it = sorted_namespaces.begin();
        it != sorted_namespaces.end(); it++) {
-    code += "/**\n * @const\n * @namespace\n */\n";
-    if (it->find('.') == std::string::npos) {
-      code += "var ";
-      if(parser_.opts.use_goog_js_export_format) {
-        exports += "goog.exportSymbol('" + *it + "', " + *it + ");\n";
-      } else {
-        exports += "this." + *it + " = " + *it + ";\n";
+    if (ts_) {
+      if (it->find('.') == std::string::npos) {
+        code += "import { flatbuffers } from \"./flatbuffers\"\n";
```

Comment:
The issue here is that flatbuffers has a .js import but no .d.ts file. I have one locally, and there are more floating around on the web, but I don't know if I should include it in this PR. Probably so. Nevertheless, that's why the flatbuffers import is optional.